src/public/app/widgets/containers/flex_container.js

```diff
@@ -10,4 +10,10 @@ export default class FlexContainer extends Container {
 
         this.attrs.style = `display: flex; flex-direction: ${direction};`;
     }
+
+    withFullSize() {
+        this.attrs.style += `width: 100%; height: 100%;`;
```

Comment:
This does not look correct.

The root container is already 100% height, banner has certain fixed height and then the rest (the main app) should be filled with flex-grow (`filling()` method)

Alternatively, wouldn't it be better to have this floating with absolute coordinates above (in the z-index sense) the main content? Since with your approach the main app will jump up and down as connection is lost/renewed.

In such case bootstrap toasts might be used as an existing solution.

Reply:
> Alternatively, wouldn't it be better to have this floating with absolute coordinates above (in the z-index sense) the main content? Since with your approach the main app will jump up and down as connection is lost/renewed.

I don't think this is a good idea. This is what it looks like:

![image](https://user-images.githubusercontent.com/50424412/143770266-918c47c7-f211-46c5-bef0-866e70ed643d.png)

Reply:
What about this:

![image](https://user-images.githubusercontent.com/617641/143788661-cf5d474b-f325-42c8-b9d7-b701cc6fc8ac.png)

can be done with few lines of code:

```javascript
toastService.showPersistent({
    id: "network-down",
    title: "Network is down!",
    message: "Trilium backend is currently not accessible.",
    icon: "wifi-off"
})
```

Reply:
That's intrusive. You can't see the full title anymoe and some buttons are harder to reach. If you're on a plane, editing your notes with only partial internet, this would drive you crazy since you wouldn't be able to do your workflow properly.

Reply:
When you're on a plane and don't have internet, basically no functionality will work correctly. You should in fact stop editing until you have an internet connection, otherwise you risk data loss.

Reply:
Wait - just for clarification. Does Trilium Notes save the progres offline and then uploads it once it has a connection back?

Reply:
Web version - no, it has no offline storage.

Desktop version works fully offline and then synchronizes with the server periodically.

Reply:
Okay in that case your solution would fit as well for the web version (only). However, we should probably change the location to the top right (or something less intrusive).